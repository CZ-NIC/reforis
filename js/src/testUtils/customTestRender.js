/*
 * Copyright (C) 2019 CZ.NIC z.s.p.o. (http://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

/* eslint import/export: "off" */

import React from 'react';
import {UIDReset} from 'react-uid';
import {StaticRouter} from 'react-router';
import {render} from '@testing-library/react'

function Wrapper({children}) {
    return <StaticRouter>
        <UIDReset>
            {children}
        </UIDReset>
    </StaticRouter>
}

const customTestRender = (ui, options) => render(ui, {wrapper: Wrapper, ...options});

// re-export everything
export * from '@testing-library/react'

// override render method
export {customTestRender as render}
