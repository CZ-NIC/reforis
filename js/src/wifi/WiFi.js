/*
 * Copyright (C) 2019 CZ.NIC z.s.p.o. (http://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

import React from "react";
import PropTypes from "prop-types";

import { ForisForm } from "foris";
import API_URLs from "common/API";

import WiFiForm from "./WiFiForm";
import ResetWiFiSettings from "./ResetWiFiSettings";

const validator = (formData) => {
    const formErrors = formData.devices.map(
        (device) => {
            if (!device.enabled) return {};

            const errors = {};
            if (device.SSID.length > 32) errors.SSID = _("SSID can't be longer than 32 symbols");
            if (device.SSID.length === 0) errors.SSID = _("SSID can't be empty");

            if (device.password.length < 8) errors.password = _("Password must contain at least 8 symbols");

            if (!device.guest_wifi.enabled) return errors;

            const guest_wifi_errors = {};
            if (device.guest_wifi.SSID.length > 32) guest_wifi_errors.SSID = _("SSID can't be longer than 32 symbols");
            if (device.guest_wifi.SSID.length === 0) guest_wifi_errors.SSID = _("SSID can't be empty");

            if (device.guest_wifi.password.length < 8) guest_wifi_errors.password = _("Password must contain at least 8 symbols");

            if (guest_wifi_errors.SSID || guest_wifi_errors.password) {
                errors.guest_wifi = guest_wifi_errors;
            }
            return errors;
        },
    );
    return JSON.stringify(formErrors) === "[{},{}]" ? null : formErrors;
};

WiFi.propTypes = {
    ws: PropTypes.object.isRequired,
};

export default function WiFi({ ws }) {
    return (
        <>
            <h1>{_("Wi-Fi")}</h1>
            <p>
                {_(`
If you want to use your router as a Wi-Fi access point, enable Wi-Fi here and fill in an SSID (the name of the access 
point) and a corresponding password. You can then set up your mobile devices, using the QR code available within the form.
            `)}
            </p>
            <ForisForm
                ws={ws}
                forisConfig={{
                    endpoint: API_URLs.wifi,
                    wsModule: "wifi",
                }}
                prepData={prepData}
                prepDataToSubmit={prepDataToSubmit}
                validator={validator}
            >
                <WiFiForm />
            </ForisForm>
            <ResetWiFiSettings ws={ws} />
        </>
    );
}

function prepData(formData) {
    formData.devices.forEach((device, idx) => {
        formData.devices[idx].channel = device.channel.toString();
    });
    return formData;
}

function prepDataToSubmit(formData) {
    formData.devices.forEach((device, idx) => {
        delete device.available_bands;

        formData.devices[idx].channel = parseInt(device.channel);

        if (!device.enabled) {
            formData.devices[idx] = { id: device.id, enabled: false };
            return;
        }

        if (!device.guest_wifi.enabled) formData.devices[idx].guest_wifi = { enabled: false };
    });
    return formData;
}
