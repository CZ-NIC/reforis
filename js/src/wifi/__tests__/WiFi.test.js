/*
 * Copyright (C) 2019 CZ.NIC z.s.p.o. (http://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

import React from 'react';
import {fireEvent, render, wait} from 'customTestRender';

import {mockedWS} from 'mockWS';
import {wifiSettingsFixture} from './__fixtures__/wifiSettings';
import mockAxios from 'jest-mock-axios';

import WiFi from '../WiFi';
import diffSnapshot from "snapshot-diff";

describe('<WiFi/>', () => {
    let firstRender;
    let getAllByText;
    let getAllByLabelText;
    let getByText;
    let asFragment;

    beforeEach(async () => {
        const mockWebSockets = new mockedWS();
        const renderRes = render(<WiFi ws={mockWebSockets}/>);
        asFragment = renderRes.asFragment;
        getAllByText = renderRes.getAllByText;
        getAllByLabelText = renderRes.getAllByLabelText;
        getByText = renderRes.getByText;
        mockAxios.mockResponse({data: wifiSettingsFixture()});
        await wait(() => renderRes.getByText('Module 1'));
        firstRender = renderRes.asFragment()
    });

    it('Snapshot both modules disabled.', () => {
        expect(firstRender).toMatchSnapshot();
    });

    it('Snapshot one module enabled.', () => {
        fireEvent.click(getAllByText('Enable')[0]);
        expect(diffSnapshot(firstRender, asFragment())).toMatchSnapshot();
    });

    it('Snapshot 2.4 GHz', () => {
        fireEvent.click(getAllByText('Enable')[0]);
        const enabledRender = asFragment();
        fireEvent.click(getAllByText('2.4')[0]);
        expect(diffSnapshot(enabledRender, asFragment())).toMatchSnapshot();
    });

    it('Snapshot guest network.', () => {
        fireEvent.click(getAllByText('Enable')[0]);
        const enabledRender = asFragment();
        fireEvent.click(getAllByText('Enable Guest Wifi')[0]);
        expect(diffSnapshot(enabledRender, asFragment())).toMatchSnapshot();
    });


    it('Post form: both modules disabled.', () => {
        fireEvent.click(getByText('Save'));
        expect(mockAxios.post).toBeCalled();
        const data = {
            devices: [
                {enabled: false, id: 0},
                {enabled: false, id: 1}
            ]
        };
        expect(mockAxios.post).toHaveBeenCalledWith('/api/wifi', data, expect.anything());
    });

    it('Post form: one module enabled.', () => {
        fireEvent.click(getAllByText('Enable')[0]);

        fireEvent.click(getByText('Save'));
        expect(mockAxios.post).toBeCalled();
        const data = {
            devices: [
                {
                    SSID: 'TestSSID1',
                    channel: 60,
                    enabled: true,
                    guest_wifi: {'enabled': false},
                    hidden: false,
                    htmode: 'HT40',
                    hwmode: '11a',
                    id: 0,
                    password: 'TestPass'
                },
                {enabled: false, id: 1}
            ]
        };
        expect(mockAxios.post).toHaveBeenCalledWith('/api/wifi', data, expect.anything());
    });

    it('Post form: 2.4 GHz', () => {
        fireEvent.click(getAllByText('Enable')[0]);
        fireEvent.click(getAllByText('2.4')[0]);

        fireEvent.click(getByText('Save'));
        expect(mockAxios.post).toBeCalled();
        const data = {
            devices: [
                {
                    SSID: 'TestSSID1',
                    channel: 0,
                    enabled: true,
                    guest_wifi: {'enabled': false},
                    hidden: false,
                    htmode: 'HT40',
                    hwmode: '11g',
                    id: 0,
                    password: 'TestPass'
                },
                {enabled: false, id: 1}
            ]
        };
        expect(mockAxios.post).toHaveBeenCalledWith('/api/wifi', data, expect.anything());
    });

    it('Post form: guest network.', () => {
        fireEvent.click(getAllByText('Enable')[0]);
        fireEvent.click(getAllByText('Enable Guest Wifi')[0]);
        fireEvent.change(getAllByLabelText('Password')[1], {target: {value: 'test_password'}});

        fireEvent.click(getByText('Save'));
        expect(mockAxios.post).toBeCalled();
        const data = {
            devices: [
                {
                    SSID: 'TestSSID1',
                    channel: 60,
                    enabled: true,
                    guest_wifi: {
                        SSID: 'TestGuestSSID',
                        enabled: true,
                        password: 'test_password'
                    },
                    hidden: false,
                    htmode: 'HT40',
                    hwmode: '11a',
                    id: 0,
                    password: 'TestPass'
                },
                {enabled: false, 'id': 1}
            ]
        };
        expect(mockAxios.post).toHaveBeenCalledWith('/api/wifi', data, expect.anything());
    });
});
