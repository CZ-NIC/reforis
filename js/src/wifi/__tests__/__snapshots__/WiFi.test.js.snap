// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WiFi/> Snapshot 2.4 GHz 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -259,207 +259,95 @@
            value=\\"0\\"
          >
            auto
          </option>
          <option
-           value=\\"36\\"
+           value=\\"1\\"
          >
            
-                         36
-                         (5180 MHz )
+                         1
+                         (2412 MHz )
                      
          </option>
          <option
-           value=\\"40\\"
+           value=\\"2\\"
          >
            
-                         40
-                         (5200 MHz )
+                         2
+                         (2417 MHz )
                      
          </option>
          <option
-           value=\\"44\\"
+           value=\\"3\\"
          >
            
-                         44
-                         (5220 MHz )
+                         3
+                         (2422 MHz )
                      
          </option>
          <option
-           value=\\"48\\"
-         >
-           
-                         48
-                         (5240 MHz )
-                     
-         </option>
-         <option
-           value=\\"52\\"
-         >
-           
-                         52
-                         (5260 MHz  ,DFS)
-                     
-         </option>
-         <option
-           value=\\"56\\"
-         >
-           
-                         56
-                         (5280 MHz  ,DFS)
-                     
-         </option>
-         <option
-           value=\\"60\\"
+           value=\\"4\\"
          >
            
-                         60
-                         (5300 MHz  ,DFS)
+                         4
+                         (2427 MHz )
                      
          </option>
          <option
-           value=\\"64\\"
+           value=\\"5\\"
          >
            
-                         64
-                         (5320 MHz  ,DFS)
+                         5
+                         (2432 MHz )
                      
          </option>
          <option
-           value=\\"100\\"
+           value=\\"6\\"
          >
            
-                         100
-                         (5500 MHz  ,DFS)
+                         6
+                         (2437 MHz )
                      
          </option>
          <option
-           value=\\"104\\"
+           value=\\"7\\"
          >
            
-                         104
-                         (5520 MHz  ,DFS)
+                         7
+                         (2442 MHz )
                      
          </option>
          <option
-           value=\\"108\\"
+           value=\\"8\\"
          >
            
-                         108
-                         (5540 MHz  ,DFS)
+                         8
+                         (2447 MHz )
                      
          </option>
          <option
-           value=\\"112\\"
+           value=\\"9\\"
          >
            
-                         112
-                         (5560 MHz  ,DFS)
+                         9
+                         (2452 MHz )
                      
          </option>
          <option
-           value=\\"116\\"
+           value=\\"10\\"
          >
            
-                         116
-                         (5580 MHz  ,DFS)
+                         10
+                         (2457 MHz )
                      
          </option>
          <option
-           value=\\"120\\"
+           value=\\"11\\"
          >
            
-                         120
-                         (5600 MHz  ,DFS)
-                     
-         </option>
-         <option
-           value=\\"124\\"
-         >
-           
-                         124
-                         (5620 MHz  ,DFS)
-                     
-         </option>
-         <option
-           value=\\"128\\"
-         >
-           
-                         128
-                         (5640 MHz  ,DFS)
-                     
-         </option>
-         <option
-           value=\\"132\\"
-         >
-           
-                         132
-                         (5660 MHz  ,DFS)
-                     
-         </option>
-         <option
-           value=\\"136\\"
-         >
-           
-                         136
-                         (5680 MHz  ,DFS)
-                     
-         </option>
-         <option
-           value=\\"140\\"
-         >
-           
-                         140
-                         (5700 MHz  ,DFS)
-                     
-         </option>
-         <option
-           value=\\"144\\"
-         >
-           
-                         144
-                         (5720 MHz  ,DFS)
-                     
-         </option>
-         <option
-           value=\\"149\\"
-         >
-           
-                         149
-                         (5745 MHz )
-                     
-         </option>
-         <option
-           value=\\"153\\"
-         >
-           
-                         153
-                         (5765 MHz )
-                     
-         </option>
-         <option
-           value=\\"157\\"
-         >
-           
-                         157
-                         (5785 MHz )
-                     
-         </option>
-         <option
-           value=\\"161\\"
-         >
-           
-                         161
-                         (5805 MHz )
-                     
-         </option>
-         <option
-           value=\\"165\\"
-         >
-           
-                         165
-                         (5825 MHz )
+                         11
+                         (2462 MHz )
                      
          </option>
        </select>
      </div>
      <div"
`;

exports[`<WiFi/> Snapshot both modules disabled. 1`] = `
<DocumentFragment>
  <h1>
    Wi-Fi
  </h1>
  <p>
    
If you want to use your router as a Wi-Fi access point, enable Wi-Fi here and fill in an SSID (the name of the access 
point) and a corresponding password. You can then set up your mobile devices, using the QR code available within the form.
            
  </p>
  <form>
    <h3>
      Wi-Fi 1
    </h3>
    <div
      class="col-sm-12 offset-lg-1 col-lg-10"
      style="margin-bottom: 1rem;"
    >
      <div
        class="custom-control custom-checkbox"
        style="margin-bottom: 0px;"
      >
        <input
          class="custom-control-input"
          id="1"
          type="checkbox"
        />
        <label
          class="custom-control-label"
          for="1"
        >
          Enable
        </label>
      </div>
    </div>
    <h3>
      Wi-Fi 2
    </h3>
    <div
      class="col-sm-12 offset-lg-1 col-lg-10"
      style="margin-bottom: 1rem;"
    >
      <div
        class="custom-control custom-checkbox"
        style="margin-bottom: 0px;"
      >
        <input
          class="custom-control-input"
          id="2"
          type="checkbox"
        />
        <label
          class="custom-control-label"
          for="2"
        >
          Enable
        </label>
      </div>
    </div>
    <button
      class="btn btn-primary  offset-lg-8 col-lg-3 col-sm-12"
      type="submit"
    >
        Save
    </button>
  </form>
  <form>
    <h4>
      Reset Wi-Fi Settings
    </h4>
    <p>
      
If a number of wireless cards doesn't match, you may try to reset the Wi-Fi settings. Note that this will remove the
current Wi-Fi configuration and restore the default values.
        
    </p>
    <button
      class="btn btn-warning offset-lg-8 col-lg-3 col-sm-12"
      type="button"
    >
        Reset Wi-Fi Settings
    </button>
  </form>
</DocumentFragment>
`;

exports[`<WiFi/> Snapshot guest network. 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -491,10 +491,89 @@
          access the internet, but aren't allowed to access other devices and the configuration interface of the router.
          Parameters of the guest network can be set in the Guest network tab.
          
        </small>
      </div>
+     <div
+       class=\\"form-group col-sm-12 offset-lg-1 col-lg-10\\"
+     >
+       <label
+         for=\\"20\\"
+       >
+         SSID
+       </label>
+       <div
+         class=\\"input-group\\"
+       >
+         <input
+           class=\\"form-control\\"
+           id=\\"20\\"
+           type=\\"text\\"
+           value=\\"TestGuestSSID\\"
+         />
+         <div
+           class=\\"input-group-append\\"
+         >
+           <button
+             class=\\"input-group-text\\"
+             type=\\"button\\"
+           >
+             <img
+               alt=\\"QR\\"
+               src=\\"/static/reforis/imgs/QR_icon.svg\\"
+               style=\\"opacity: 0.67;\\"
+               width=\\"20\\"
+             />
+           </button>
+         </div>
+       </div>
+     </div>
+     <div
+       class=\\"form-group col-sm-12 offset-lg-1 col-lg-10\\"
+     >
+       <label
+         for=\\"21\\"
+       >
+         Password
+       </label>
+       <div
+         class=\\"input-group\\"
+       >
+         <input
+           autocomplete=\\"new-password\\"
+           class=\\"form-control  is-invalid\\"
+           id=\\"21\\"
+           required=\\"\\"
+           type=\\"password\\"
+           value=\\"\\"
+         />
+         <div
+           class=\\"input-group-append\\"
+         >
+           <button
+             class=\\"input-group-text\\"
+             type=\\"button\\"
+           >
+             <i
+               class=\\"fa fa-eye\\"
+             />
+           </button>
+         </div>
+       </div>
+       <div
+         class=\\"invalid-feedback\\"
+       >
+         Password must contain at least 8 symbols
+       </div>
+       <small
+         class=\\"form-text text-muted\\"
+       >
+         
+         WPA2 pre-shared key, that is required to connect to the network.
+     
+       </small>
+     </div>
      <h3>
        Wi-Fi 2
      </h3>
      <div
        class=\\"col-sm-12 offset-lg-1 col-lg-10\\"
@@ -517,10 +596,11 @@
          </label>
        </div>
      </div>
      <button
        class=\\"btn btn-primary  offset-lg-8 col-lg-3 col-sm-12\\"
+       disabled=\\"\\"
        type=\\"submit\\"
      >
          Save
      </button>
    </form>"
`;

exports[`<WiFi/> Snapshot one module enabled. 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -31,10 +31,470 @@
          >
            Enable
          </label>
        </div>
      </div>
+     <div
+       class=\\"form-group col-sm-12 offset-lg-1 col-lg-10\\"
+     >
+       <label
+         for=\\"4\\"
+       >
+         SSID
+       </label>
+       <div
+         class=\\"input-group\\"
+       >
+         <input
+           class=\\"form-control\\"
+           id=\\"4\\"
+           required=\\"\\"
+           type=\\"text\\"
+           value=\\"TestSSID1\\"
+         />
+         <div
+           class=\\"input-group-append\\"
+         >
+           <button
+             class=\\"input-group-text\\"
+             type=\\"button\\"
+           >
+             <img
+               alt=\\"QR\\"
+               src=\\"/static/reforis/imgs/QR_icon.svg\\"
+               style=\\"opacity: 0.67;\\"
+               width=\\"20\\"
+             />
+           </button>
+         </div>
+       </div>
+     </div>
+     <div
+       class=\\"form-group col-sm-12 offset-lg-1 col-lg-10\\"
+     >
+       <label
+         for=\\"5\\"
+       >
+         Password
+       </label>
+       <div
+         class=\\"input-group\\"
+       >
+         <input
+           autocomplete=\\"new-password\\"
+           class=\\"form-control\\"
+           id=\\"5\\"
+           required=\\"\\"
+           type=\\"password\\"
+           value=\\"TestPass\\"
+         />
+         <div
+           class=\\"input-group-append\\"
+         >
+           <button
+             class=\\"input-group-text\\"
+             type=\\"button\\"
+           >
+             <i
+               class=\\"fa fa-eye\\"
+             />
+           </button>
+         </div>
+       </div>
+       <small
+         class=\\"form-text text-muted\\"
+       >
+         
+         WPA2 pre-shared key, that is required to connect to the network.
+     
+       </small>
+     </div>
+     <div
+       class=\\"col-sm-12 offset-lg-1 col-lg-10\\"
+       style=\\"margin-bottom: 1rem;\\"
+     >
+       <div
+         class=\\"custom-control custom-checkbox\\"
+         style=\\"margin-bottom: 0px;\\"
+       >
+         <input
+           class=\\"custom-control-input\\"
+           id=\\"6\\"
+           type=\\"checkbox\\"
+         />
+         <label
+           class=\\"custom-control-label\\"
+           for=\\"6\\"
+           style=\\"margin-bottom: 0px;\\"
+         >
+           Hide SSID
+         </label>
+       </div>
+       <small
+         class=\\"form-text text-muted\\"
+       >
+         If set, network is not visible when scanning for available networks.
+       </small>
+     </div>
+     <div
+       class=\\"form-group col-sm-12 offset-lg-1 col-lg-10\\"
+       style=\\"margin-bottom: 1rem;\\"
+     >
+       <label
+         class=\\"col-12\\"
+         for=\\"7\\"
+         style=\\"padding-left: 0px;\\"
+       >
+         GHz
+       </label>
+       <div
+         class=\\"custom-control custom-radio custom-control-inline\\"
+       >
+         <input
+           class=\\"custom-control-input\\"
+           id=\\"hwmode-0-0\\"
+           name=\\"hwmode-0\\"
+           type=\\"radio\\"
+           value=\\"11g\\"
+         />
+         <label
+           class=\\"custom-control-label\\"
+           for=\\"hwmode-0-0\\"
+         >
+           2.4
+         </label>
+       </div>
+       <div
+         class=\\"custom-control custom-radio custom-control-inline\\"
+       >
+         <input
+           checked=\\"\\"
+           class=\\"custom-control-input\\"
+           id=\\"hwmode-0-1\\"
+           name=\\"hwmode-0\\"
+           type=\\"radio\\"
+           value=\\"11a\\"
+         />
+         <label
+           class=\\"custom-control-label\\"
+           for=\\"hwmode-0-1\\"
+         >
+           5
+         </label>
+       </div>
+       <small
+         class=\\"form-text text-muted\\"
+       >
+         
+         The 2.4 GHz band is more widely supported by clients, but tends to have more interference. The 5 GHz band is a
+         newer standard and may not be supported by all your devices. It usually has less interference, but the signal
+         does not carry so well indoors.
+       </small>
+     </div>
+     <div
+       class=\\"form-group col-sm-12 offset-lg-1 col-lg-10\\"
+     >
+       <label
+         for=\\"8\\"
+       >
+         802.11n/ac mode
+       </label>
+       <select
+         class=\\"custom-select\\"
+         id=\\"8\\"
+       >
+         <option
+           value=\\"NOHT\\"
+         >
+           Disabled
+         </option>
+         <option
+           value=\\"HT20\\"
+         >
+           802.11n - 20 MHz wide channel
+         </option>
+         <option
+           value=\\"HT40\\"
+         >
+           802.11n - 40 MHz wide channel
+         </option>
+         <option
+           value=\\"VHT20\\"
+         >
+           802.11ac - 20 MHz wide channel
+         </option>
+         <option
+           value=\\"VHT40\\"
+         >
+           802.11ac - 40 MHz wide channel
+         </option>
+         <option
+           value=\\"VHT80\\"
+         >
+           802.11ac - 80 MHz wide channel
+         </option>
+       </select>
+       <small
+         class=\\"form-text text-muted\\"
+       >
+         
+         Change this to adjust 802.11n/ac mode of operation. 802.11n with 40 MHz wide channels can yield higher
+         throughput but can cause more interference in the network. If you don't know what to choose, use the default
+         option with 20 MHz wide channel.
+     
+       </small>
+     </div>
+     <div
+       class=\\"form-group col-sm-12 offset-lg-1 col-lg-10\\"
+     >
+       <label
+         for=\\"9\\"
+       >
+         Channel
+       </label>
+       <select
+         class=\\"custom-select\\"
+         id=\\"9\\"
+       >
+         <option
+           value=\\"0\\"
+         >
+           auto
+         </option>
+         <option
+           value=\\"36\\"
+         >
+           
+                         36
+                         (5180 MHz )
+                     
+         </option>
+         <option
+           value=\\"40\\"
+         >
+           
+                         40
+                         (5200 MHz )
+                     
+         </option>
+         <option
+           value=\\"44\\"
+         >
+           
+                         44
+                         (5220 MHz )
+                     
+         </option>
+         <option
+           value=\\"48\\"
+         >
+           
+                         48
+                         (5240 MHz )
+                     
+         </option>
+         <option
+           value=\\"52\\"
+         >
+           
+                         52
+                         (5260 MHz  ,DFS)
+                     
+         </option>
+         <option
+           value=\\"56\\"
+         >
+           
+                         56
+                         (5280 MHz  ,DFS)
+                     
+         </option>
+         <option
+           value=\\"60\\"
+         >
+           
+                         60
+                         (5300 MHz  ,DFS)
+                     
+         </option>
+         <option
+           value=\\"64\\"
+         >
+           
+                         64
+                         (5320 MHz  ,DFS)
+                     
+         </option>
+         <option
+           value=\\"100\\"
+         >
+           
+                         100
+                         (5500 MHz  ,DFS)
+                     
+         </option>
+         <option
+           value=\\"104\\"
+         >
+           
+                         104
+                         (5520 MHz  ,DFS)
+                     
+         </option>
+         <option
+           value=\\"108\\"
+         >
+           
+                         108
+                         (5540 MHz  ,DFS)
+                     
+         </option>
+         <option
+           value=\\"112\\"
+         >
+           
+                         112
+                         (5560 MHz  ,DFS)
+                     
+         </option>
+         <option
+           value=\\"116\\"
+         >
+           
+                         116
+                         (5580 MHz  ,DFS)
+                     
+         </option>
+         <option
+           value=\\"120\\"
+         >
+           
+                         120
+                         (5600 MHz  ,DFS)
+                     
+         </option>
+         <option
+           value=\\"124\\"
+         >
+           
+                         124
+                         (5620 MHz  ,DFS)
+                     
+         </option>
+         <option
+           value=\\"128\\"
+         >
+           
+                         128
+                         (5640 MHz  ,DFS)
+                     
+         </option>
+         <option
+           value=\\"132\\"
+         >
+           
+                         132
+                         (5660 MHz  ,DFS)
+                     
+         </option>
+         <option
+           value=\\"136\\"
+         >
+           
+                         136
+                         (5680 MHz  ,DFS)
+                     
+         </option>
+         <option
+           value=\\"140\\"
+         >
+           
+                         140
+                         (5700 MHz  ,DFS)
+                     
+         </option>
+         <option
+           value=\\"144\\"
+         >
+           
+                         144
+                         (5720 MHz  ,DFS)
+                     
+         </option>
+         <option
+           value=\\"149\\"
+         >
+           
+                         149
+                         (5745 MHz )
+                     
+         </option>
+         <option
+           value=\\"153\\"
+         >
+           
+                         153
+                         (5765 MHz )
+                     
+         </option>
+         <option
+           value=\\"157\\"
+         >
+           
+                         157
+                         (5785 MHz )
+                     
+         </option>
+         <option
+           value=\\"161\\"
+         >
+           
+                         161
+                         (5805 MHz )
+                     
+         </option>
+         <option
+           value=\\"165\\"
+         >
+           
+                         165
+                         (5825 MHz )
+                     
+         </option>
+       </select>
+     </div>
+     <div
+       class=\\"col-sm-12 offset-lg-1 col-lg-10\\"
+       style=\\"margin-bottom: 1rem;\\"
+     >
+       <div
+         class=\\"custom-control custom-checkbox\\"
+         style=\\"margin-bottom: 0px;\\"
+       >
+         <input
+           class=\\"custom-control-input\\"
+           id=\\"10\\"
+           type=\\"checkbox\\"
+         />
+         <label
+           class=\\"custom-control-label\\"
+           for=\\"10\\"
+           style=\\"margin-bottom: 0px;\\"
+         >
+           Enable Guest Wifi
+         </label>
+       </div>
+       <small
+         class=\\"form-text text-muted\\"
+       >
+         
+         Enables Wi-Fi for guests, which is separated from LAN network. Devices connected to this network are allowed to
+         access the internet, but aren't allowed to access other devices and the configuration interface of the router.
+         Parameters of the guest network can be set in the Guest network tab.
+         
+       </small>
+     </div>
      <h3>
        Wi-Fi 2
      </h3>
      <div
        class=\\"col-sm-12 offset-lg-1 col-lg-10\\""
`;
